File: Java/romanToInt.java
import java.util.HashMap;
import java.util.Map;  

class Solution  {

    public int romanToInt (String s) {

        Map<Character, Integer> values = new HashMap<>();

        values.put('I', 1);
        values.put('V', 5);
        values.put('X', 10);
        values.put('L', 50);
        values.put('C', 100);
        values.put('D', 500);
        values.put('M', 1000);


        int finalValue = 0;
        int preValue = 0;

        for(int i = s.length()-1; i >= 0; i--) {
            int currentValue = values.get(s.charAt(i));
            if (currentValue < preValue) {
                finalValue -= currentValue;
            } else {
                finalValue += currentValue;
            }

            preValue = currentValue;
        }



        return finalValue;


    }
}

File: C#/romanToInt.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int RomanToInt(string s) 
    {
        Dictionary<char, int> values = new Dictionary<char, int>
        {
            {'I', 1}, {'V', 5}, {'X', 10},
            {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}
        };

        int finalValue = 0;
        int preValue = 0;

        for (int i = s.Length - 1; i >= 0; i--) 
        {
            char currentChar = s[i];
            if (!values.TryGetValue(currentChar, out int currentValue)) 
            {
                throw new ArgumentException($"Invalid Roman numeral character: {currentChar}");
            }

            if (currentValue < preValue) 
            {
                finalValue -= currentValue;
            } 
            else 
            {
                finalValue += currentValue;
            }

            preValue = currentValue;
        }

        return finalValue;
    }
}
